completions/ip

```diff
@@ -357,7 +357,23 @@ _comp_cmd_ip()
         n | neigh | neighbor | neighbour)
             case $subcmd in
                 add | del | change | replace)
-                    # TODO
+                    case $prev in
+                        lladdr)
+                            _comp_compgen_mac_addresses
+                            ;;
+                        nud)
+                            _comp_cmd_ip__neigh_states "$1"
+                            ;;
+                        dev)
+                            _comp_compgen_available_interfaces
+                            ;;
+                        proxy)
```

Comment:
`protocol` takes an arg, should we try and complete it or explicitly "non-complete" it?

Reply:
Made it complete using `/etc/iproute2/rt_protos`